### PR TITLE
Tests 

### PR DESCRIPTION
**Pull request by maximelaval**
Added some test for the CastVote and Election vote class .
Created test for CreateLao,UpdateLao, StateLao protocol. 
Refactored tests for Lao model class. 
Deleted the test fragments for this branch , and put them in a separate branch for a future PR.

### CODE REVIEW

File: fe2-android/app/src/test/java/com/github/dedis/student20_pop/model/LaoTest.java [hunk 1]
```diff
@@ -1,172 +1,253 @@
...
+    private static final String electionId2 = "rollCallId2";
+    private static final String electionId3 = "rollCallId3";
+    private static final Set<String> WITNESSES = new HashSet<>(Arrays.asList("0x3434", "0x4747"));
+    private static final Set<String> WITNESSES_WITH_NULL = new HashSet<>(Arrays.asList("0x3939", null, "0x4747"));
+
+    private static final Lao LAO_1 = new Lao(LAO_1_ID, LAO_NAME_1);
+    private static final Map<String, RollCall> rollCalls = new HashMap<String, RollCall>() {{
+        put(rollCallId1, new RollCall());
+        put(rollCallId2, new RollCall());
+        put(rollCallId3, new RollCall());
```
Comment (celinecamacho) [inline fe2-android/app/src/test/java/com/github/dedis/student20_pop/model/LaoTest.java:31]: The class RollCall doesn't have a default constructor, either implement one or create a roll call using the existing constructor. You have this issue multiple times on this file

File: fe2-android/app/src/androidTest/java/com/github/dedis/student20_pop/home/LaunchFragment2.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+import static androidx.test.espresso.matcher.ViewMatchers.withClassName;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.anything;
+import static org.hamcrest.Matchers.is;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class LaunchFragment2 {
```
Comment (celinecamacho) [inline fe2-android/app/src/androidTest/java/com/github/dedis/student20_pop/home/LaunchFragment2.java:41]: You can delete the deprecated launch fragment test and rename this one `LaunchFragmentTest`

File: fe2-android/app/src/test/java/com/github/dedis/student20_pop/model/LaoTest.java (hunk 1 above)
Comment (maximelaval) [reply fe2-android/app/src/test/java/com/github/dedis/student20_pop/model/LaoTest.java:31]: I don't really understand why you mean by that, isn't this the default constructor in RollCall that I use  : public RollCall() {
    this.attendees = new HashSet<>();
  }
?

Comment (celinecamacho) [reply fe2-android/app/src/test/java/com/github/dedis/student20_pop/model/LaoTest.java:31]: I see, you have the old RollCall constructor, you should merge with master

**Review by celinecamacho — APPROVED** [review]

**Pull request merged**